hr_period/models/hr_fiscal_year.py

```diff
@@ -0,0 +1,328 @@
+
+strptime = datetime.strptime
+strftime = datetime.strftime
+
+INTERVALS = {
+    'annually': (relativedelta(months=12), 1),
+    'semi-annually': (relativedelta(months=6), 2),
+    'quarterly': (relativedelta(months=3), 4),
+    'bi-monthly': (relativedelta(months=2), 6),
+    'semi-monthly': (relativedelta(weeks=2), 24),
+    'monthly': (relativedelta(months=1), 12),
+    'bi-weekly': (relativedelta(weeks=2), 26),
+    'weekly': (relativedelta(weeks=1), 52),
+    'daily': (relativedelta(days=1), 365),
+}
+
+
+@api.model
+def get_schedules(self):
+    return [
+        ('annually', _('Annually (1)')),
+        ('semi-annually', _('Semi-annually (2)')),
+        ('quarterly', _('Quarterly (4)')),
+        ('bi-monthly', _('Bi-monthly (6)')),
+        ('monthly', _('Monthly (12)')),
+        ('semi-monthly', _('Semi-monthly (24)')),
+        ('bi-weekly', _('Bi-weekly (26)')),
+        ('weekly', _('Weekly (52)')),
+        ('daily', _('Daily (365)')),
+    ]
+
+
+@api.model
+def get_payment_days(self):
+    expr = _('%s day of the next period')
+    expr_2 = _('%s day of the current period')
+    return [
+        ('1', expr % _('First')),
+        ('2', expr % _('Second')),
+        ('3', expr % _('Third')),
+        ('4', expr % _('Fourth')),
+        ('5', expr % _('Fifth')),
+        ('0', expr_2 % _('Last')),
+    ]
+
+
+class HrFiscalYear(models.Model):
+    _name = 'hr.fiscalyear'
+    _description = 'HR Fiscal Year'
+
+    @api.model
+    def _default_date_start(self):
+        today = datetime.now()
+        return datetime(today.year, 1, 1).strftime(DF)
+
+    @api.model
+    def _default_date_stop(self):
+        today = datetime.now()
+        return datetime(today.year, 12, 31).strftime(DF)
+
+    name = fields.Char(
+        'Fiscal Year',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        'Company',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda obj: obj.env.user.company_id
+    )
+    date_start = fields.Date(
+        'Start Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="The first day of the first period of the fiscal year.",
+        default=_default_date_start
+    )
+    date_stop = fields.Date(
+        'End Date',
+        required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="The last day of the last period of the fiscal year.",
+        default=_default_date_stop
+    )
+    period_ids = fields.One2many(
+        'hr.period',
+        'fiscalyear_id',
+        'Periods',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('open', 'Open'),
+            ('done', 'Closed'),
```

Comment:
I think we should have a mothod to close the fiscalyear, if a company decide at the middle of the year to change the policy of payment on payslip, we will have to go on each period linked to close it, so the fiscalyear to compute the close state, or probably to have a date untill the fiscalyear is active, and to recompute on contracts if a new fiscalyear is defined